src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.props

```diff
@@ -88,6 +88,7 @@ The .NET Foundation licenses this file to you under the MIT license.
       <LinkerArg Include="-static" Condition="'$(StaticallyLinked)' == 'true'" />
       <LinkerArg Include="-lgssapi_krb5" Condition="'$(TargetOS)' != 'OSX' and '$(StaticallyLinked)' != 'true'" />
       <LinkerArg Include="-lrt" Condition="'$(TargetOS)' != 'OSX'" />
+      <LinkerArg Include="-lssl -lcrypto" Condition="'$(TargetOS)' != 'OSX'" />
```

Comment:
Does this add new prerequisites that are needed for NativeAOT compilation?

https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/docs/prerequisites.md#ubuntu-1604

Reply:
This is a good question and I don't really have an answer. After our conversation, it sounds like there are two builds of runtime here: the source build and the packaged build. If people get the packaged build for linux, then these libraries will be loaded via dlopen. If they're using the source build, then the libraries need to be explicitly passed.

I'm not sure how to divide these situations -- should we make openssl dev libraries required always even though it may not be explicitly required? I'm leaning slightly towards that position.

Reply:
Is it always going to be `-lssl` or can it be `-lssl<some suffix>` on some distros? It may be best to disable NativeAOT for crossgen2 in source build for now and deal with it separately.

Reply:
The source build looks for the right ssl version to link against here: https://github.com/dotnet/runtime/blob/main/src/native/libs/System.Security.Cryptography.Native/extra_libs.cmake#L7

Reply:
The building works fine, it's only when publishing crossgen2 that NativeAOT has to link the resulting binaries together, and if the NativeAOT binaries reference openssl directly, then they expect to be passed the SOs during linking.

I could simply disable NativeAOT of crossgen during the source build, but that would still mean that the NativeAOT binaries that are constructed during source build will not actually work with the NativeAOT targets, because the linking command is wrong.

I could try to disable NativeAOT building for source build entirely... but that seems like we're headed in the wrong direction. We'll almost certainly need to eventually get it working, and then we'll still have to solve this problem.

Reply:
I agree that we need to find the right solution for source build to deal with this issue. My suggestion is to deal with it separately - open an issue on it and disable it in source build for now to unblock this PR.